packages/mui-material/src/AvatarGroup/AvatarGroup.js

```diff
@@ -142,6 +138,10 @@ const AvatarGroup = React.forwardRef(function AvatarGroup(inProps, ref) {
       as={component}
       ownerState={ownerState}
       className={clsx(classes.root, className)}
+      style={{
+        '--AvatarGroup-spacing': marginValue ? `${marginValue}px` : undefined,
+        ...other.style,
+      }}
```

Comment:
I think this won't fix the problem? the `{…other}` below will override the `style` here?

It should be:

```js
ref={ref}
{…other}
style={{
  '--AvatarGroup-spacing': marginValue ? `${marginValue}px` : undefined,
  ...other.style,
}}
```

Reply:
Good catch. Fixed https://github.com/mui/material-ui/pull/44208/commits/64fceb6239ec441e32453895697bcb3fd10c0b03 It'd have failed if someone passed a `styled` prop.